exporter/collector/config.go

```diff
@@ -69,58 +66,19 @@ type AttributeMapping struct {
-	// If a metric belongs to one of these domains it does not get a prefix.
-	KnownDomains []string `mapstructure:"known_domains"`
-
-	// If true, set the instrumentation_source and instrumentation_version
-	// labels. Defaults to true.
-	InstrumentationLibraryLabels bool `mapstructure:"instrumentation_library_labels"`
-
-	// If true, this will send all timeseries using `CreateServiceTimeSeries`.
-	// Implicitly, this sets `SkipMetricDescriptor` to true.
-	CreateServiceTimeSeries bool `mapstructure:"create_service_timeseries"`
-
-	// Buffer size for the channel which asynchronously calls CreateMetricDescriptor. Default
-	// is 10.
-	CreateMetricDescriptorBufferSize int `mapstructure:"create_metric_descriptor_buffer_size"`
-
-	// If true, the exporter will copy OTel's service.name, service.namespace, and
-	// service.instance.id resource attributes into the GCM timeseries metric labels. This
-	// option is recommended to avoid writing duplicate timeseries against the same monitored
-	// resource. Disabling this option does not prevent resource_filters from adding those
-	// labels. Default is true.
-	ServiceResourceLabels bool `mapstructure:"service_resource_labels"`
-
-	// If provided, resource attributes matching any filter will be included in metric labels.
-	// Defaults to empty, which won't include any additional resource labels. Note that the
-	// service_resource_labels option operates independently from resource_filters.
-	ResourceFilters []ResourceFilter `mapstructure:"resource_filters"`
-
-	// CumulativeNormalization normalizes cumulative metrics without start times or with
-	// explicit reset points by subtracting subsequent points from the initial point.
-	// It is enabled by default. Since it caches starting points, it may result in
-	// increased memory usage.
-	CumulativeNormalization bool `mapstructure:"cumulative_normalization"`
-
-	// This enables calculation of an estimated sum of squared deviation.  It isn't correct,
-	// so we don't send it by default, and don't expose it to users. For some uses, it is
-	// expected, however.
-	EnableSumOfSquaredDeviation bool `mapstructure:"sum_of_squared_deviation"`
-
-	// GetMetricName is not settable in config files, but can be used by other
-	// exporters which extend the functionality of this exporter. It allows
-	// customizing the naming of metrics. baseName already includes type
-	// suffixes for summary metrics, but does not (yet) include the domain prefix
-	GetMetricName func(baseName string, metric pmetric.Metric) (string, error)
-
-	// MapMonitoredResource is not exposed as an option in the configuration, but
-	// can be used by other exporters to extend the functionality of this
-	// exporter. It allows overriding the function used to map otel resource to
-	// monitored resource.
-	MapMonitoredResource func(pcommon.Resource) *monitoredrespb.MonitoredResource
+	GetMetricName                    func(baseName string, metric pmetric.Metric) (string, error)
```

Comment:
Not sure about your rules, but this may be a breaking change if someone initializes a struct without using named member arguments.

Reply:
Yes, we do consider that a breaking change.  The exporter/collector package is currently unstable (and is only used by collector-contrib to my knowledge), so this is acceptable for now.  I called that out more clearly in the description